File: test/_3_tree/java/ElementTest.java
package _3_tree.java;

import org.junit.Test;

import static org.junit.Assert.*;

public class ElementTest {

    @Test
    public void testSearch() throws Exception {
        Tree<Integer, String> tree = new Tree<>(2, "b", new Leaf<>(1, "a"), new Leaf<>(3, "c"));
        assertEquals("a", tree.binarySearch(1));
        assertEquals("b", tree.binarySearch(2));
        assertEquals("c", tree.binarySearch(3));
        Tree<String, String> complex = new Tree<>("i", "j",
                new Tree<>("k", "l", new Leaf<>("a", "b"), new Leaf<>("c", "d")),
                new Tree<>("m", "n", new Leaf<>("e", "f"), new Leaf<>("g", "h")));
        assertEquals("f", complex.search("e"));
        assertEquals("n", complex.search("m"));
        assertEquals("j", complex.search("i"));
    }
}

File: test/_4_tree/kotlin/ElementTest.kt
package _4_tree.kotlin

import _4_tree.kotlin.Element.Leaf
import _4_tree.kotlin.Element.Tree
import org.junit.Test

import org.junit.Assert.*

class ElementTest {

    @Test
    fun testSearch() {
        val tree = Tree(2, "b", Leaf(1, "a"), Leaf(3, "c"))
        assertEquals("a", tree.binarySearch(1))
        assertEquals("b", tree.binarySearch(2))
        assertEquals("c", tree.binarySearch(3))
        val complex = Tree("i", "j",
                           Tree("k", "l", Leaf("a", "b"), Leaf("c", "d")),
                           Tree("m", "n", Leaf("e", "f"), Leaf("g", "h")))
        assertEquals("f", complex.search("e"))
        assertEquals("n", complex.search("m"))
        assertEquals("j", complex.search("i"))
    }
}